### ISSUE TITLE
Screen Readers should read month & year

### ISSUE THREAD

**Issue by tylerlinquata** [jquense/react-big-calendar#2488]
### Clear and concise description of the problem

When using a Screen Reader, the user will never hear the month or year that are in the toolbar. When tabbing though date cells, only the day is read. This has been flagged as an accessibility issue by my company.

Current experience using VoiceOver:

https://github.com/jquense/react-big-calendar/assets/25713879/92689545-99da-452a-bda8-e8b71fc8bade

### Suggested solution

In `DateHeader.js`, we can add a label that allows the screen reader to read the full localized date. This would be visually hidden. 

Example js:
```
<button
  type="button"
  className="rbc-button-link"
  onClick={onDrillDown}
  role="cell"
>
  <span aria-hidden="true">{label}</span>
  <span className="rbc-sr-only">
    {localizer.format(date, 'ddd MMM DD YYYY')}
  </span>
</button>
```

Example css:
```
.rbc-sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
}
```
Example video:

https://github.com/jquense/react-big-calendar/assets/25713879/96adf020-0e43-44c0-837d-e030b306acf8

### Alternative

One alternative would be to change the `Toolbar` such that it is read by the screen reader. This may be considered compliant from an accessibility standpoint, but it doesn't feel like a great user experience.

### Additional context

I'm happy to contribute this fix if allowed.

### Validations

- [X] Read the [docs](http://jquense.github.io/react-big-calendar/examples/index.html).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Would you like to open a PR for this feature?

- [X] I'm willing to open a PR

**Comment by cutterbl** [comment]
From an accessibility standpoint though, is that really the right way to do it? We do pass the `date` into the `DateHeader` component. Perhaps if we wrapped the `label` in an html `<time>` tag with a formatted `datetime` attribute?

Accessibility is a large problem in Big Calendar, and improvements are on our radar for the 'next' version. We'd be happy to look at any PR you might submit for this particular issue, and welcome your feedback in our Slack community's 'next' channel on how to better approach accessibility going forward.

**Comment by tylerlinquata** [comment]
@cutterbl I'm not entirely sure what the "correct" solution would be here, in my experience it has been hard to find any sort of definitive "this is correct" vs "this is not correct" in regard to accessibility outside of the WCAG spec, which isn't a practical guide.

As far as using a `<time>` tag, afaik it isn't supported by screen readers and will be ignored.  The `aria-hidden` trick seems to be a widely used way to get the screen reader to read text that is more descriptive than what is displayed on the screen.
